integration/spark/shared/src/main/java/io/openlineage/spark/agent/facets/builder/CustomEnvironmentFacetBuilder.java

```diff
@@ -0,0 +1,108 @@
+/*
+/* Copyright 2018-2023 contributors to the OpenLineage project
+/* SPDX-License-Identifier: Apache-2.0
+*/
+
+package io.openlineage.spark.agent.facets.builder;
+
+import io.openlineage.spark.agent.facets.EnvironmentFacet;
+import io.openlineage.spark.api.CustomFacetBuilder;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.BiConsumer;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.ArrayUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.spark.scheduler.SparkListenerJobStart;
+
+/**
+ * {@link CustomFacetBuilder} that generates a {@link EnvironmentFacet} when using OpenLineage on
+ * Databricks.
+ */
+@Slf4j
+public class CustomEnvironmentFacetBuilder
+    extends CustomFacetBuilder<SparkListenerJobStart, EnvironmentFacet> {
+  private Map<String, Object> envProperties;
+
+  public static boolean isCustomEnvironmentVarCollectionEnabled() {
+    return System.getenv().containsKey("CUSTOM_VARS");
```

Comment:
I think longer form `CUSTOM_ENV_VARIABLES` would be preferred, even if more annoying to type. Or even `CUSTOM_OPENLINEAGE_ENV_VARIABLES`. Not strongly attached to this though.

Reply:
Also, I think this is something that should also be specified as Spark config, as there is rest of our configuration settings.

For example, as `spark.openlineage.facets.custom_env_vars`

Reply:
Yeap that makes sense. I am now passing it in as a spark config.